File: src/main/java/leetcode/Problem167.java
package leetcode;

@Difficulty(Level.EASY)
@Algorithms(Algorithm.TWO_POINTERS)
@BeatsPercent(100)
@TimeComplexity(worst = Complexity.LINEAR_N)
@SpaceComplexity(worst = Complexity.CONSTANT)
public class Problem167 {
    public int[] twoSum(int[] numbers, int target) {
        int lo = 0;
        int hi = numbers.length - 1;
        while (lo < hi) {
            int diff = target - (numbers[lo] + numbers[hi]);
            if (diff == 0) {
                return new int[] {lo + 1, hi + 1};
            }
            if (diff > 0) {
                lo++;
            } else {
                hi--;
            }
        }
        throw new RuntimeException();
    }
}

File: src/problem_167.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        lo, hi = 0, len(numbers) - 1
        while lo < hi:
            diff = numbers[lo] + numbers[hi] - target
            if diff == 0:
                return [lo + 1, hi + 1]
            elif diff > 0:
                hi -= 1
            else:
                lo += 1
